daphne/src/roles.rs

```diff
@@ -575,8 +586,15 @@ where
         match req.media_type {
             Some(MEDIA_TYPE_AGG_INIT_REQ) => {
                 let agg_init_req = AggregateInitializeReq::get_decoded(&req.payload)?;
+                let mut first_metadata: Option<&ReportMetadata> = None;
+                if !agg_init_req.report_shares.is_empty() {
+                    first_metadata = Some(&agg_init_req.report_shares[0].metadata);
+                }
```

Comment:
[first()](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.first) is more idiomatic (not sure if the syntax is precisely correct)

```suggestion
                let first_metadata = agg_init_req.report_shares.first().map(|report_share| report_share.metadata);
```

Reply:
it was almost right, needed an & :). fixed, thanks!